PWGEM/PhotonMeson/Tasks/gammaConversions.cxx

```diff
@@ -494,18 +602,73 @@ struct GammaConversions {
     }
   }
 
+bool RVsZPassLineCut(float v0radius, float z)
+{
+  if(TMath::Abs(z) < LineCutZ0 + v0radius * LineCutZRSlope)
+  {
+    return true;
+  }
+  return false;
+}
+
+  //this is really ugly but it works. Would be easier if .end() worked..
+  template <typename TTRACKS>
+  void fillhpeDivpGammaHistograms(mapStringHistPtr& theContainer, TTRACKS const& theTwoV0Daughters)
+  {
+    float p_electron[2];
+    Int_t counter = 0;
+
+    for (auto& lTrack : theTwoV0Daughters)
+    {
+      p_electron[counter] = lTrack.p();
+      counter++;
+    }
+
+    fillTH1(theContainer, "hpeDivpGamma", p_electron[0]/(p_electron[0]+p_electron[1]));
+    fillTH1(theContainer, "hpeDivpGamma", p_electron[1]/(p_electron[0]+p_electron[1]));
+  }
+
+  template <typename TV0>
+  void fillRVsZHistograms(mapStringHistPtr& theContainer, TV0 const& theV0)
+  {
+    if(RVsZPassLineCut(theV0.v0radius(), theV0.z()) == true)
+    { 
+      fillTH2(theContainer, "hRVsZ", theV0.v0radius(), theV0.z());
+    }
+  }
+
   template <typename TV0>
   void fillV0Histograms(mapStringHistPtr& theContainer, TV0 const& theV0, float const& theV0CosinePA)
   {
     fillTH1(theContainer, "hEta", theV0.eta());
     fillTH1(theContainer, "hPhi", theV0.phi());
     fillTH1(theContainer, "hPt", theV0.pt());
     fillTH1(theContainer, "hConvPointR", theV0.v0radius());
+    fillTH1(theContainer, "hConvPointZ",  theV0.z());
     fillTH1(theContainer, "hCosPAngle", theV0CosinePA);
     fillTH2(theContainer, "hArmenteros", theV0.alpha(), theV0.qtarm());
     fillTH2(theContainer, "hPsiPt", theV0.psipair(), theV0.pt());
   }
 
+  template <typename TTRACKS>
+  void fillV0Histograms_recalculated(mapStringHistPtr& theContainer, TTRACKS const& theTwoV0Daughters)
+  {
+    auto firstV0Daughter = theTwoV0Daughters.begin();
+    TVector3 lConvPoint(firstV0Daughter.recalculatedVtxX(), firstV0Daughter.recalculatedVtxY(), firstV0Daughter.recalculatedVtxZ());
+
+    fillTH1(theContainer, "hConvPointR_recalc", lConvPoint.Perp());
+    fillTH1(theContainer, "hConvPointZ_recalc", firstV0Daughter.recalculatedVtxZ());
```

Comment:
this quantity I would add as dynamic column to the V0DaughterTracks table in gammaTables.h
And then put the fill calls into fillV0Histograms